src/libraries/Microsoft.PowerFx.Core/Public/Config/VersionHash.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.PowerFx.Core;
+using Microsoft.PowerFx.Core.App;
+using Microsoft.PowerFx.Core.Binding;
+using Microsoft.PowerFx.Core.Binding.BindInfo;
+using Microsoft.PowerFx.Core.Entities;
+using Microsoft.PowerFx.Core.Functions;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Types.Enums;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx
+{
+    /// <summary>
+    /// Version hash. Can tell if a value has changed.  
+    /// The only thing this can be used for is to tell if a snapshot has changed. 
+    /// A version stamp should only be compared to a previous version of the same stamp. 
+    /// </summary>
+    [DebuggerDisplay("Version:{_value}")]
+    public struct VersionHash
+    {
+        private int _value;
+
+        private VersionHash(int value)
+        {
+            _value = value;
+        }
+
+        public static VersionHash New()
+        {
+            // Give psuedo-random seed.
+            // This can help protect against 2 objects accidentally appearing the same. 
+            var value = (int)DateTime.UtcNow.Ticks;
+            return new VersionHash(value);
+        }
+
+        /// <summary>
+        /// Note that object has changed. 
+        /// </summary>
+        public void Inc()
+        {
+            _value++;
+        }
+
+        public VersionHash Combine(VersionHash other)
+        {
+            return new VersionHash(HashCombine(_value, other._value));
+        }
+
+        internal static int HashCombine(int a, int b)
```

Comment:
Should these methods be internal? When would a host need to create or increment their own hash?

Also, does this class need to be public, or should it also be internal?

Reply:
They probably could all be internal.  Can always make public later. 
Just needs to be public if a host wants to tell if a symbol table has changed, but I don't think they need to do that.